File: static/code/sample/daisylink/gdl-sedistance-b.cs
// In this sample:
// Update distance in cm every second

using GHIElectronics.DUELink;
using System;

var availablePort = DUELinkController.GetConnectionPort();
var duelink = new DUELinkController(availablePort);

float GetDistance() {
    var ret = duelink.Engine.ExecuteCommand("Distance()");
    return ret;
}

while (true) {
    var distance = GetDistance();
    Console.WriteLine($"Distance: {distance} cm");
    Thread.Sleep(1000);
}

File: static/code/sample/daisylink/gdl-sedistance-b.py
# In this sample:
# Update distance in cm every second

import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

def GetDistance():
    ret = duelink.Engine.ExecuteCommand("Distance()")
    return ret

while True:
    distance = GetDistance()
    print(f"Distance: {distance} cm")
    time.sleep(1)